src/Models/PathEditModel.vala

```diff
@@ -68,103 +70,164 @@ public class Akira.Models.PathEditModel : Object {
         update_view ();
     }
 
-    public void add_live_points_to_path (Geometry.Point[] live_pts, Lib.Modes.PathEditMode.Type live_command, int length) {
-        commands.resize (commands.length + 1);
-        commands[commands.length - 1] = live_command;
-
+    public void add_live_points_to_path (Utils.PathSegment live_pts) {
```

Comment:
I haven't read all this in detail. But my feeling is that this class has way more code than it should for what it does. I'm not sure if it's code duplication or something in the PathSegment struct that requires extra code.